streams/src/test/java/org/apache/kafka/streams/kstream/PrintedTest.java

```diff
@@ -0,0 +1,123 @@
+import org.apache.kafka.test.TestUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+public class PrintedTest {
+
+    private final ByteArrayOutputStream sysOut = new ByteArrayOutputStream();
+    private final Printed<String, Integer> sysOutPrinter = Printed.toSysOut();
+
+    @Before
+    public void before() {
+        System.setOut(new PrintStream(sysOut));
+    }
+
+    @After
+    public void after() {
+        System.setOut(null);
+    }
+
+    @Test
+    public void shouldCreateProcessorThatPrintsToFile() throws IOException {
+        final File file = TestUtils.tempFile();
+        final ProcessorSupplier<String, Integer> processorSupplier = Printed.<String, Integer>toFile(file.getPath()).build("processor");
+        final Processor<String, Integer> processor = processorSupplier.get();
+        processor.process("hi", 1);
+        processor.close();
+        try (final FileInputStream stream = new FileInputStream(file)) {
+            final byte[] data = new byte[stream.available()];
+            stream.read(data);
+            assertThat(new String(data, StandardCharsets.UTF_8.name()), equalTo("[processor]: hi, 1\n"));
+        }
+    }
+
+    @Test
+    public void shouldCreateProcessorThatPrintsToStdOut() throws UnsupportedEncodingException {
+        final ProcessorSupplier<String, Integer> supplier = sysOutPrinter.build("processor");
+        supplier.get().process("good", 2);
+        assertThat(new String(sysOut.toByteArray(), StandardCharsets.UTF_8.name()), equalTo("[processor]: good, 2\n"));
+    }
+
+    @Test
+    public void shouldPrintWithLabel() throws UnsupportedEncodingException {
+        final Processor<String, Integer> processor = sysOutPrinter
+                .withLabel("label")
+                .build("processor")
+                .get();
+
+        processor.process("hello", 3);
+        assertThat(new String(sysOut.toByteArray(), StandardCharsets.UTF_8.name()), equalTo("[label]: hello, 3\n"));
+    }
+
+    @Test
+    public void shouldPrintWithKeyValueMapper() throws UnsupportedEncodingException {
+        final Processor<String, Integer> processor = sysOutPrinter.withKeyValueMapper(new KeyValueMapper<String, Integer, String>() {
+            @Override
+            public String apply(final String key, final Integer value) {
+                return String.format("%s -> %d", key, value);
+            }
+        }).build("processor")
+                .get();
+        processor.process("hello", 1);
+        assertThat(new String(sysOut.toByteArray(), StandardCharsets.UTF_8.name()), equalTo("[processor]: hello -> 1\n"));
```

Comment:
nit: replace the duplicate `new String(..)` calls with a private method `getString(byte[])`